vscripts/confogl_hittable_control.nut

```diff
@@ -0,0 +1,74 @@
+local isGauntletFinale = false;
+local allowDamageTimestampTable = {};
+local hittableDamage = 100;
+local hittableImmunityTime = 1.2;
+
+ // finale_start doesn't fire for Gauntlet finales, but this does. We can't use OnGameplayStart(),
+ // because the trigger_finale does not exist until after the initial radio conversation.
+function OnGameEvent_finale_radio_start(params)
+{
+    if (NetProps.GetPropInt(Entities.FindByClassname(null, "trigger_finale"), "m_type") == 1)
+    {
+        isGauntletFinale = true;
+    }
+}
+
+function AllowTakeDamage(damageTable)
+{
+    if (!damageTable.Attacker.IsValid() || !damageTable.Inflictor.IsValid() || !damageTable.Victim.IsValid())
+    {
+        return true;
+    }
+    else if (!damageTable.Attacker.IsPlayer() || !damageTable.Victim.IsPlayer())
+    {
+        return true;
+    }
+
+    if (damageTable.Inflictor != null)
+    {
+        local inflictorClassname = damageTable.Inflictor.GetClassname();
+        if (inflictorClassname == "prop_physics" || inflictorClassname == "prop_car_alarm")
+        {
+            if (damageTable.Attacker.GetZombieType() == 8) // Attacker is a Tank.
+            {
+                if (damageTable.Attacker == damageTable.Victim)
+                {
+                    return false; // Prevent Tank self-harm.
+                }
+                else if (damageTable.Victim.IsSurvivor())
+                {
+                    if (isGauntletFinale)
+                    {
+                        damageTable.DamageDone = hittableDamage * 4; // Hittable damage on Gauntlet finales is divided by 4 past where this function returns.
+                    }
+                    else
+                    {
+                        damageTable.DamageDone = hittableDamage;
+                    }
+                    if (damageTable.Victim in allowDamageTimestampTable)
+                    {
+                        if ((allowDamageTimestampTable[damageTable.Victim] - Time()) > 0.0)
```

Comment:
I've not experienced any unexpected behavior since squashing a bug where the immunity time would only work once per survivor, but it could probably still use more rigorous testing. I'll try at least throwing a print in there and see if anything weird comes out. I'm trying to avoid worldspawn or any props etc. being stored in the table.